DLCS.Repository.Tests/Assets/AssetRepositoryTests.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Features.Image.Requests;
+using DLCS.Model.Assets;
+using DLCS.Repository.Assets;
+using DLCS.Repository.Caching;
+using FluentAssertions;
+using LazyCache.Mocks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Test.Helpers.Integration;
+using Xunit;
+
+namespace DLCS.Repository.Tests.Assets
+{
+    [Trait("Category", "Database")]
+    [Collection(DatabaseCollection.CollectionName)]
+    public class AssetRepositoryTests
+    {
+        private readonly DlcsContext dbContext;
+        private readonly IAssetRepository sut;
+
+        public AssetRepositoryTests(DlcsDatabaseFixture dbFixture)
+        {
+            dbContext = dbFixture.DbContext;
+            // We want this turned on to match live behaviour
+            dbContext.ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.TrackAll;
```

Comment:
Might be best to new up a DbContext here and pass that to the sut rather than changing the one on the `DlcsDatabaseFixture`. 
`DlcsDatabaseFixture` is a collection fixture so it'll be shared across all classes in this same assembly. Settings `TrackAll` will set it for all other test classes too, which might lead to unexpected behaviour.

Reply:
All works nicely... but my test `AssetRepository_Throws_if_Same_Asset_Tracked` now fails, because it no longer throws an exception! It's testing that you're not using the same dbcontext to make changes to tracked assets - but now the tests are using a different dbcontext to the API.